### PR TITLE
change to gdal 2.0.2

### PR DESCRIPTION
**Pull request by kmuehlbauer**
add has_geos check

I did not check if the patches (osx, win) also apply to the 2.0.2 version yet.

### CODE REVIEW

**Comment by pelson** [comment]
Cool! Well that is easier to review! :smile:

File: recipe/run_test.py [hunk 1]
```diff
@@ -5,3 +5,19 @@
...
+    pnt2 = ogr.CreateGeometryFromWkt( 'POINT(30 20)' )
+    ogrex = ogr.GetUseExceptions()
+    ogr.DontUseExceptions()
+    hasgeos = pnt1.Union( pnt2 ) is not None
+    if ogrex:
+        ogr.UseExceptions()
+    return hasgeos
+
+if has_geos():
+  print('GDAL has GEOS compiled in')
```
Comment (pelson) [inline recipe/run_test.py]: Suggest these are made assertions. Do you want GEOS in there, or not?

Comment (kmuehlbauer) [inline recipe/run_test.py]: Yes, I need GEOS, the test was suggested by @ocefpaf in the other pr. I admit I did not use `assert` much until now.

Do you mean something like:

`assert has_geos(),  'GEOS not available within GDAL'`

Comment (ocefpaf) [inline recipe/run_test.py]: :+1: Let's just raise an error if geos support is not there.

**Comment by ocefpaf** [comment]
@kmuehlbauer add your GitHub handle at the of the recipe to become a maintainer.

```yaml
 extra:
   recipe-maintainers:
     - ocefpaf
     - kmuehlbauer 
```

**Comment by pelson** [comment]
Looks like the patch doesn't quite work on this version for Windows...

**Comment by kmuehlbauer** [comment]
The patch is for 1.11.4, should we try without patch next time.

**Comment by kmuehlbauer** [comment]
It seems that this is fixed in 2.0.2 https://trac.osgeo.org/gdal/ticket/6255

**Comment by ocefpaf** [comment]
> It seems that this is fixed in 2.0.2 https://trac.osgeo.org/gdal/ticket/6255

Yep.  I "stole" that from the 2.0.2 version.  You can drop the `MSC_VER_1900_snprintf.patch`.

**Comment by kmuehlbauer** [comment]
@ocefpaf @pelson removing the patch did the trick. The first appveyor windows job has green. 

Also linux works with python27. So waiting for the other jobs to finish. osx doesn't work at all.

**Comment by ocefpaf** [comment]
> So waiting for the other jobs to finish. osx doesn't work at all.

This is new...

```
install_name_tool -add_rpath @loader_path/../lib/ /Users/travis/miniconda3/envs/_build/bin/gdal_contour
install_name_tool -delete_rpath /Users/travis/miniconda3/envs/_build/lib /Users/travis/miniconda3/envs/_build/bin/gdal_contour
Skipping -delete_rpath, file doesn't contain that LC_RPATH
An unexpected error has occurred, please consider sending the
following traceback to the conda GitHub issue tracker at:
    https://github.com/conda/conda-build/issues
Include the output of the command 'conda info' in your report.
Traceback (most recent call last):
  File "/Users/travis/miniconda3/bin/conda-build", line 5, in <module>
    sys.exit(main())
  File "/Users/travis/miniconda3/lib/python3.4/site-packages/conda_build/main_build.py", line 194, in main
    args_func(args, p)
  File "/Users/travis/miniconda3/lib/python3.4/site-packages/conda_build/main_build.py", line 474, in args_func
    args.func(args, p)
  File "/Users/travis/miniconda3/lib/python3.4/site-packages/conda_build/main_build.py", line 423, in execute
    include_recipe=args.include_recipe)
  File "/Users/travis/miniconda3/lib/python3.4/site-packages/conda_build/build.py", line 496, in build
    post_build(m, sorted(files2 - files1))
  File "/Users/travis/miniconda3/lib/python3.4/site-packages/conda_build/post.py", line 322, in post_build
    mk_relative(m, f)
  File "/Users/travis/miniconda3/lib/python3.4/site-packages/conda_build/post.py", line 291, in mk_relative
    mk_relative_osx(path)
  File "/Users/travis/miniconda3/lib/python3.4/site-packages/conda_build/post.py", line 266, in mk_relative_osx
    assert_relative_osx(path)
  File "/Users/travis/miniconda3/lib/python3.4/site-packages/conda_build/post.py", line 279, in assert_relative_osx
    assert not name.startswith(config.build_prefix), path
AssertionError: /Users/travis/miniconda3/envs/_build/bin/gdal_contour
The command "conda build ./recipe" exited with 1.
```

I will do some googling to see what is happening.

**Comment by pelson** [comment]
Yeah, that's an interesting one. @msarahan - seen this before?

**Comment by kmuehlbauer** [comment]
Uh oh, seems this issue persists for the windows x86 python35 version or is this something new:

```
  Creating library gdal_i.lib and object gdal_i.exp
odbccp32.lib(dllload.obj) : error LNK2019: unresolved external symbol __vsnwprintf_s referenced in function _StringCchPrintfW
gdal200.dll : fatal error LNK1120: 1 unresolved externals
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\link.EXE"' : return code '0x460'
Stop.
 
C:\conda\conda-bld\work\gdal-2.0.2>if errorlevel 1 exit 1 
Command failed: C:\windows\system32\cmd.exe /c call bld.bat
Command exited with code 1
```

**Comment by kmuehlbauer** [comment]
@ocefpaf How is this going further? It would be great to have gdal 2.0.2 available within conda-forge, at least the working versions.

**Comment by ocefpaf** [comment]
> @ocefpaf How is this going further? It would be great to have gdal 2.0.2 available within conda-forge, at least the working versions.

Not sure.  The AppVeyor failure is expected, but the OS X is not.  One test we could do is to avoid updating conda in this [line](https://github.com/conda-forge/gdal-feedstock/blob/master/.travis.yml#L30) and see what happens.

**Comment by kmuehlbauer** [comment]
@ocefpaf OK, let's give it a try. But no need to have the other two OS run again. What is the simplest way to deactivate CircleCi and appveyor?

**Comment by ocefpaf** [comment]
I never used them before but I know that adding the message `[skip appveyor]` will skip AppVyeor and `[ci skip]`  skip all CIs. Not sure if there is `[skip travis]` and/or `[skip circle]`.  I am googling to see if I can find something.

**Comment by kmuehlbauer** [comment]
@ocefpaf Your suggestion worked. All osx jobs compiled well, as far as I can see. Don't know why the appveyor doesn't come back. It also seemed to work with this skip line.

Can I do something more on my end?

File: .travis.yml [hunk 2]
```diff
@@ -27,7 +27,7 @@ install:
 
       export PATH=/Users/travis/miniconda3/bin:$PATH
 
-      conda update --yes conda
+      # conda update --yes conda
```
Comment (ocefpaf) [inline .travis.yml]: We need to check the logs as a newer version that fix the bugs was released right after the buggy one.  If we can be with the latest we should not comment this. (At least now we know it was a conda bug.)

**Comment by ocefpaf** [comment]
> @ocefpaf Your suggestion worked. All osx jobs compiled well, as far as I can see.

Phew! I was running out of ideas.

> Don't know why the appveyor doesn't come back. It also seemed to work with this skip line.

AppVeyor is slow and we are hammering it pretty hard lately. There is nothing we can do but wait. However, since we know AppVeyor was green before (with some expected failures) we don't need to wait.

> Can I do something more on my end?

Not really.  I guess I need to merge this onto a branch.  Let me finish some things here and I will get back to this. (If I forget please ping me again.)

**Comment by kmuehlbauer** [comment]
@ocefpaf Thanks! I'll come back if something happens here, or ping you on monday. Happy weekend.

**Comment by kmuehlbauer** [comment]
ping @ocefpaf let me know, if I can do anything to help getting this done.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
+    md5: 573865f3f59ba7b4f8f4cddf223b52a5
     patches:
         - clang.patch  # [osx]
         - prepend-dll.patch  # [win]
-        - MSC_VER_1900_snprintf.patch  # [win]
         - install_scripts.patch
 
 build:
     number: 0
+    skip: True # [not osx]
```
Comment (ocefpaf) [inline recipe/meta.yaml]: @kmuehlbauer Why are we skipping all but `osx`? If I recall correctly Windows will fail only on PY35 (that is OK), but Linux was build correctly, right?

**Comment by kmuehlbauer** [comment]
@ocefpaf That was to only check the osx build. I left that for the time being, because if I change it now, all CI will run again. 

Shall I fix it now?

**Comment by ocefpaf** [comment]
> Shall I fix it now?

Yes. We need to remove that to merge this.

**Comment by ocefpaf** [comment]
@kmuehlbauer I created the [gdal2](https://github.com/conda-forge/gdal-feedstock/tree/gdal2) branch. Not sure if you can re-direct you PR to that branch or if I can do it with some git magic.

**Comment by kmuehlbauer** [comment]
@ocefpaf No, unfortunately I cannot redirect the pullrequest. I suggest to let me pull the new gdal2 branch and make the changes there.

**Comment by ocefpaf** [comment]
> I suggest to let me pull the new gdal2 branch and make the changes there.

:+1: 

Sorry to make you do that. Next time I'll create the branch first.

**Comment by kmuehlbauer** [comment]
pr switched to branch [gdal2](https://github.com/conda-forge/gdal-feedstock/tree/gdal2) #6 

closing this

**Pull request closed without merging**